pyk/src/pyk/klean/k2lean4.py

```diff
@@ -200,10 +200,20 @@ def _symbol_ctor(self, sort: str, symbol: str) -> Ctor:
     def _symbol_ident(symbol: str) -> str:
         if symbol.startswith('Lbl'):
             symbol = symbol[3:]
-        symbol = unmunge(symbol)
-        if not _VALID_LEAN_IDENT.fullmatch(symbol):
-            symbol = f'«{symbol}»'
-        return symbol
+        return K2Lean4._escape_ident(symbol, kore=True)
+
+    @staticmethod
+    def _var_ident(var: str) -> str:
+        assert var.startswith('Var')
+        return K2Lean4._escape_ident(var[3:], kore=True)
+
+    @staticmethod
+    def _escape_ident(ident: str, *, kore: bool = False) -> str:
+        if kore:
+            ident = unmunge(ident)
```

Comment:
In what scenario would `kore = False`?

Reply:
If `ident` is not a KORE identifier, in particular, for rule labels.